### PR TITLE
Cargo-binutils: add explanation

### LINKED ISSUE #122166: Document how to use cargo-binutils ?
@Stupremee  sorry to bother you with this. Is there a way to document how to use cargo-binutils ?
I added it to may `buildInputs` inside `mkShell`, but `cargo size` still says
```
Failed to execute tool: size
No such file or directory (os error 2)
```

**To Reproduce**
1. add `cargo-binutils` inside a buildInputs of a mkShell
2. call `cargo size`

**Expected behavior**
no error output

**Screenshots**

**Additional context**
let me know if I'm doing something wrong here.
Maybe the ideal thing would be to add something in the description?

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by happysalada**
###### Motivation for this change
cargo-binutils doesn't work out of the box. This adds a small explanation on what is needed to make it work

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

@Stupremee Let me know what you think.

closes #122166

### CODE REVIEW

File: pkgs/development/tools/rust/cargo-binutils/default.nix [hunk 1]
```diff
@@ -5,23 +5,28 @@ rustPlatform.buildRustPackage rec {
...
+      cp -R ${repo} $out
+      chmod -R +w $out
+      cp ${./Cargo.lock} $out/Cargo.lock
+    '';
 
   cargoSha256 = "sha256-Zrl269PacPi81TrGTIDzmVndgGY5i5lYyspiOj43rpw=";
 
   meta = with lib; {
-    description = "Cargo subcommands to invoke the LLVM tools shipped with the Rust toolchain";
+    description = ''
```
Comment (Stupremee) [inline pkgs/development/tools/rust/cargo-binutils/default.nix:26]: I would probably not say that `rust-overlay` is the only thing, because `nixpkgs-mozilla` can do that too. So maybe rephrase it to smth like: "In order for this to work, you either need to run `rustup component add llvm-tools-preview` or install the `llvm-tools-preview` component using your Nix library (e.g. nixpkgs-mozilla)"

Comment (Stupremee) [inline pkgs/development/tools/rust/cargo-binutils/default.nix:26]: Isn't there even a `longDescription` field? That may be more fitting

Comment (happysalada) [reply pkgs/development/tools/rust/cargo-binutils/default.nix]: Thanks a lot!

Comment (happysalada) [reply pkgs/development/tools/rust/cargo-binutils/default.nix]: You're right, that's a much better way.
